lib/schema_registry/client.rb

```diff
@@ -27,11 +28,21 @@ def initialize(code, message)
 
   class Client
 
-    attr_reader :endpoint, :username, :password
+    attr_reader :endpoint, :username, :password, :client_key, :client_certificate, :ca_certificate, :verify_mode
 
-    def initialize(endpoint, username = nil, password = nil)
+    def initialize(endpoint, username = nil, password = nil, client_certificate = nil, client_key = nil, ca_certificate = nil, verify_mode = :verify_peer)
       @endpoint = URI(endpoint)
       @username, @password = username, password
+      
+      unless client_certificate.nil? && client_key.nil?
+        @client_certificate = client_certificate
+        @client_key = client_key
+        @ca_certificate = ca_certificate
+        @verify_mode = OpenSSL::SSL.const_get(
```

Comment:
I prefer using the `OpenSSL::SSL` constants directly, rather than symbols.

Reply:
k, but I am using this from another module ( logstash ) so its harder to pass an object from the logstash conf to the logstash codec and then to your module